crates/wasmtime/src/instance.rs

```diff
@@ -160,19 +82,15 @@ impl Instance {
     /// [inst]: https://webassembly.github.io/spec/core/exec/modules.html#exec-instantiation
     /// [issue]: https://github.com/bytecodealliance/wasmtime/issues/727
     /// [`ExternType`]: crate::ExternType
-    pub fn new(store: &Store, module: &Module, imports: &[Extern]) -> Result<Instance, Error> {
-        if !Engine::same(store.engine(), module.engine()) {
-            bail!("cross-`Engine` instantiation is not currently supported");
-        }
-
-        let handle = with_imports(store, module.compiled_module(), imports, |imports| {
-            instantiate(store, module.compiled_module(), imports, Box::new(()))
-        })?;
-
-        Ok(Instance {
-            handle,
-            module: module.clone(),
-        })
+    pub fn new(store: &Store, module: &Module, imports: &[Extern]) -> Result<Instance> {
+        let allocator = store
+            .engine()
+            .config()
+            .instance_allocator
+            .as_deref()
+            .unwrap_or_else(|| &store.engine().config().default_instance_allocator);
+
+        allocator.allocate(store, module, imports)
```

Comment:
And if we just had `instance_allocator`, then this code would simplify a bit as well.

But I feel like I'm missing some kind of context about why we have both (a comment above the field definitions in the `Config` struct would really help readers out here!)

Reply:
Agreed.  I can comment on why the config is storing both, but I'm definitely open to any suggestions you might have regarding a different approach for handling internal host object instantiations vs. module instantiations.

The end goal being that I don't want a pooling allocator to treat host objects as counting towards any limits placed on the total number of instances supported by the pool.

Reply:
The config no longer stores both the configured instance allocator and the default allocator.